hooks/resolve_publish.py

```diff
@@ -0,0 +1,34 @@
+# Copyright (c) 2017 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+from sgtk import Hook
+
+
+class PublishResolver(Hook):
+    """
+    Hook used to resolve publish records in Shotgun into
+    local form on a machine.
+    """
+
+    def resolve_path(self, sg_publish_data):
+        """
+        Resolves a Shotgun publish record into a local file on disk.
+        If the method returns None, the default implementation will be used.
```

Comment:
Would it be cleaner/better to have the default logic live here in the hook (either full implementation or calling a method on core)?